pkg/controller/collector/timestamp.go

```diff
@@ -57,43 +65,41 @@ func NewTimestamp(config TimestampConfig) (*Timestamp, error) {
 }
 
 func (t *Timestamp) Collect(ch chan<- prometheus.Metric) error {
-	ctx := context.Background()
-	gvk, err := apiutil.GVKForObject(t.newRuntimeObjectFunc(), t.k8sClient.Scheme())
-	if err != nil {
-		return microerror.Mask(err)
+	list := &metav1.PartialObjectMetadataList{}
+	{
+		gvk, err := apiutil.GVKForObject(t.newRuntimeObjectListFunc(), t.scheme)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		list.SetGroupVersionKind(gvk)
 	}
-	gvk.Kind += "List"
-	list := &unstructured.UnstructuredList{}
-	list.SetGroupVersionKind(gvk)
 
-	err = t.k8sClient.CtrlClient().List(ctx, list)
+	ctx := context.Background()
+	err := t.ctrlClient.List(ctx, list, &client.ListOptions{
+		LabelSelector: t.selector,
+	})
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
 	for _, object := range list.Items {
-		if !t.selector.Matches(selector.NewLabels(object.GetLabels())) {
-			// Skip objects that do not match the selector labels
-			continue
-		}
```

Comment:
Here are the most important changes, passing in selectors for server-side filtering rather than doing it client-side.